lib/src/main/java/graphql/nadel/engine/transform/partition/NadelPartitionKeyExtractor.kt

```diff
@@ -0,0 +1,29 @@
+package graphql.nadel.engine.transform.partition
+
+import graphql.language.ScalarValue
+import graphql.schema.GraphQLInputValueDefinition
+
+/**
+ * Interface to extract the partition key from a scalar value.
+ */
+interface NadelPartitionKeyExtractor {
+    /**
+     * Function to extract the partition key from the given scalar value.
+     *
+     * A `null` return value indicates that the scalar value should not be partitioned.
+     */
+    fun getPartitionKey(
+        scalarValue: ScalarValue<*>,
+        inputValueDef: GraphQLInputValueDefinition,
+        context: NadelFieldPartitionContext,
+    ): String? {
+        return null
```

Comment:
Should this have a default? doesn't seem right

I think the `noop` version below should hardcode null, but the user should be otherwise forced to add an impl given this is the only function in this interface

Reply:
Yep makes sense 👍